### PR TITLE
New features for Elasticsearch image: S3 repository plugin and configure_s3.sh

### PR DESCRIPTION
**Pull request by Phandora**
Hi team!

This PR adds `S3 repository plugin` and `configure_s3.sh` to Elasticsearch image. This features allow us to create a S3 repository on AWS to perform backup operations.

The plugin provides a S3 repository type. The repository defaults to using ECS IAM Role or EC2 IAM Role credentials for authentication, but we recommend to add the _necessary policy_ to write on the S3 repository into the instance where Elasticsearch is hosted.

The configure_s3.sh script is launched by load_settings.sh. By default, the environment variable _ENABLE_CONFIGURE_S3_ is set to false, so it will not create any repository. To successfully create a S3 repository some environment variables are needed:

- **ENABLE_CONFIGURE_S3**: true
- **ELASTICSEARCH_IP**: elasticsearch
- **ELASTICSEARCH_PORT**: 9200
- **S3_BUCKET_NAME**:  your_bucket_name
- **S3_PATH**: path_for_bucket
- **S3_REPOSITORY_NAME**: your_s3_repository_name

Optionally, **S3_ELASTIC_MAJOR** can be set to add the specified version to the path bucket. By default, current Elasticsearch version is checked and added at the end of the path and the repository name.

Best regards,

Mayte Ariza

### CODE REVIEW

File: elasticsearch/config/configure_s3.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
```
Comment (jesuslinares) [inline elasticsearch/config/configure_s3.sh]: Add `set -e`

File: elasticsearch/config/configure_s3.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    elastic_ip_port="$2"
+    bucket_name="$3"
+    path="$4"
+    repository_name="$5"
+
+    if [ $1 == 5 ];then
+        version="$6"
+    else
+        version=`curl -s $elastic_ip_port | grep number | cut -d"\"" -f4 | cut -c1`
```
Comment (jesuslinares) [inline elasticsearch/config/configure_s3.sh]: Validate `version` in both cases (if and else)

File: elasticsearch/config/load_settings.sh [hunk 3]
```diff
@@ -23,6 +23,25 @@ done
 
 >&2 echo "Elastic is up - executing command"
 
+if [ $ENABLE_CONFIGURE_S3 ]; then
+
+  sleep 10
```
Comment (jesuslinares) [inline elasticsearch/config/load_settings.sh]: Add a comment explaining why we need this sleep.

Comment (manuasir) [reply elasticsearch/config/load_settings.sh]: ¿Is 10 seconds enough for it?

File: elasticsearch/config/configure_s3.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            "settings": {
+            "bucket": "'$bucket_name'",
+            "base_path": "'$s3_path'"
+        }
+    }
+    '
+
+}
+
+
```
Comment (manuasir) [inline elasticsearch/config/configure_s3.sh]: Please, write a comment about this function.

**Pull request merged**